scripts/build_android.sh

```diff
@@ -23,9 +23,9 @@ export JVM_OPTS="-Xmx3200m"
 export GRADLE_OPTS='-Dorg.gradle.jvmargs="-Xmx2048m -XX:+HeapDumpOnOutOfMemoryError"'
 
 # processing xml is messed up a bit after the original introduction of gradle
-# so a useless install of ant here even if using gradle/ninja
+# so require a useless install of ant here even if using gradle/ninja
 
-sudo apt-get install -y ant
+which ant > /dev/null || { echo "FATAL: ant is not installed; install manually and retry."; exit 1; }
```

Comment:
Maybe this line good be written in a way that is more compatible with generic shell syntax like:
```
if which ant >/dev/null; then
    echo "FATAL: ant is not installed; install manually and retry." >&2
    exit 1
fi
```

Reply:
This will fail if ant is found and succeed if it is not found (just tried it). I am not a shell guru, though; all I know is that if success/failure/true/false/`if` is highly confusing.

Redirecting to stderr is a good point, though.

Reply:
Mmm... OK, I missed the fact that there is an issue because of the mix of if then construct and set -e